### ISSUE TITLE
findExistingAddresses is checking dead containers

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#2815]
Because I re-used the `allContainerIDs` list we already had, but this contains live and dead containers (for a good reason).  

This doesn't stop it working; just that it would be more efficient to concentrate on the live ones when looking for IP addresses.

Here: https://github.com/weaveworks/weave/blob/f7ce16cba88a31d190727b9e5219de721554e18e/prog/weaver/main.go#L314

**Comment by dlmiddlecote** [comment]
Hi,

Is the solution here to create a new list `runningContainerIDs` from `dockerCli`, and then pass that list to `findExistingAddresses`, or something more efficient than going to the Docker API twice to get the two lists?

**Comment by bboreham** [comment]
I don't suppose there is one correct solution. 

Whilst it is true that the list coming back from Docker includes a running/exited status, it may be overly convoluted to plumb that back to the callers. 

The problem I saw when I filed this issue is that an `Inspect` call is done for every dead container, and there may conceivably be hundreds of those. One more `List` call may be an ok price for simplicity.
